tests/integ/scala/test_dataframe_join_suite.py

```diff
@@ -660,54 +669,52 @@ def test_natural_cross_joins(session):
     ]
 
 
-def test_clone_with_join_dataframe(session):  # TODO: support session.table
+@pytest.mark.localtest
+def test_clone_with_join_dataframe(session):
     table_name1 = Utils.random_name_for_temp_object(TempObjectType.TABLE)
-    try:
-        Utils.create_table(session, table_name1, "c1 int, c2 int")
-        session.sql(f"insert into {table_name1} values(1, 2), (2, 3)").collect()
-        df = session.table(table_name1)
+    session.create_dataframe([[1, 2], [2, 3]], schema=["c1", "c2"]).write.save_as_table(
+        table_name1, table_type="temporary"
+    )
 
-        assert df.collect() == [Row(1, 2), Row(2, 3)]
+    df = session.table(table_name1)
 
-        cloned_df = copy.copy(df)
-        #  Cloned DF has the same conent with original DF
-        assert cloned_df.collect() == [Row(1, 2), Row(2, 3)]
+    assert df.collect() == [Row(1, 2), Row(2, 3)]
 
-        join_df = df.join(cloned_df, df["c1"] == cloned_df["c2"])
-        assert join_df.collect() == [Row(2, 3, 1, 2)]
-        # Cloned join DF
-        cloned_join_df = copy.copy(join_df)
-        assert cloned_join_df.collect() == [Row(2, 3, 1, 2)]
+    cloned_df = copy.copy(df)
+    #  Cloned DF has the same conent with original DF
+    assert cloned_df.collect() == [Row(1, 2), Row(2, 3)]
 
-    finally:
-        Utils.drop_table(session, table_name1)
+    join_df = df.join(cloned_df, df["c1"] == cloned_df["c2"])
+    assert join_df.collect() == [Row(2, 3, 1, 2)]
+    # Cloned join DF
+    cloned_join_df = copy.copy(join_df)
+    assert cloned_join_df.collect() == [Row(2, 3, 1, 2)]
 
 
+# TODO: Fix simplifier copy
```

Comment:
The current simplifier codepath doesn't work well with `copy`, this is sort of caused by that.